File: app/src/main/java/com/grohden/niceanimals/ui/holders/GalleryViewHolder.java
package com.grohden.niceanimals.ui.holders;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.ImageView;
import android.widget.ProgressBar;

import com.grohden.niceanimals.R;
import com.grohden.niceanimals.realm.entities.NiceAnimal;
import com.squareup.picasso.Callback;
import com.squareup.picasso.Picasso;

import butterknife.BindView;
import butterknife.ButterKnife;

public class GalleryViewHolder extends RecyclerView.ViewHolder implements Callback {

    @BindView(R.id.gallery_image)
    ImageView mAnimalImage;

    @BindView(R.id.gallery_image_loader)
    ProgressBar mProgressBar;

    public GalleryViewHolder(View itemView) {
        super(itemView);
        ButterKnife.bind(this, itemView);
    }

    public void bindAnimal(NiceAnimal animal) {
        Picasso.get()
                .load(animal.getPictureUrl())
                .fit()
                .centerInside()
                .into(mAnimalImage, this);
    }

    @Override
    public void onSuccess() {
        mProgressBar.setVisibility(View.GONE);
    }

    @Override
    public void onError(Exception e) {
        mProgressBar.setVisibility(View.GONE);
    }
}

File: app/src/main/java/com/grohden/niceanimals/ui/holders/GalleryViewHolder.kt
package com.grohden.niceanimals.ui.holders

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import com.grohden.niceanimals.extensions.at
import com.grohden.niceanimals.realm.entities.NiceAnimal
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.gallery_image.view.*

class GalleryViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

    fun bindAnimal(animal: NiceAnimal) {
        Picasso.get()
            .load(animal.pictureUrl)
            .fit()
            .centerInside()
            .at(itemView.galleryImage) {
                itemView.galleryImageLoader.visibility = View.GONE
            }
    }
}
